staging/src/k8s.io/kubectl/pkg/config/v1alpha1/types.go

```diff
@@ -62,6 +65,30 @@ type Preference struct {
 	// "kubectl getn control-plane-1 --output=json" expands to "kubectl get node --output=json control-plane-1"
 	// +listType=atomic
 	Aliases []AliasOverride `json:"aliases"`
+
+	// credentialPluginPolicy specifies the policy governing which, if any, client-go
+	// credential plugins may be executed. It MUST be one of { "", "AllowAll", "DenyAll", "Allowlist" }.
+	// If the policy is "", then it falls back to "AllowAll" (this is required
+	// to maintain backward compatibility). If the policy is DenyAll, no
+	// credential plugins may run. If the policy is Allowlist, only those
+	// plugins meeting the criteria specified in the `credentialPluginAllowlist`
+	// field may run.
+	CredentialPluginPolicy clientcmdapi.PolicyType `json:"credentialPluginPolicy,omitempty"`
```

Comment:
Do we add new fields in beta also in alpha?. I was assuming that we shouldn't do this.